src/crystal/system/win32/file.cr

```diff
@@ -26,8 +26,8 @@ module Crystal::System::File
     fd
   end
 
-  def self.mktemp(prefix : String, suffix : String?, dir : String) : {LibC::Int, String}
-    path = "#{tempdir}\\#{prefix}.#{::Random::Secure.hex}#{suffix}"
+  def self.mktemp(prefix : String?, suffix : String?, dir : String) : {LibC::Int, String}
+    path = "#{Crystal::System::Dir.tempdir}\\#{prefix}.#{::Random::Secure.hex}#{suffix}"
```

Comment:
Shouldn't this be similar to the unix impl? 👇 https://github.com/crystal-lang/crystal/blob/625884b1acbbbf4db321ac76738ebef35a0aba93/src/crystal/system/unix/file.cr#L16-L18

```suggestion
  def self.mktemp(prefix, suffix, dir) : {LibC::Int, String}
    path = "#{dir}\\#{prefix}.#{::Random::Secure.hex}#{suffix}"
```

Reply:
The method signature doesn't matter much, but yes, the `dir` argument should be honoured.